src/poetry/utils/password_manager.py

```diff
@@ -240,3 +230,17 @@ def delete_http_password(self, name: str) -> None:
             self.keyring.delete_password(name, username)
 
         self._config.auth_config_source.remove_property(f"http-basic.{name}")
+
+    def get_credential(
+        self, *names: str, username: str | None = None
+    ) -> HTTPAuthCredential:
+        if self.use_keyring:
+            return self.keyring.get_credential(*names, username=username)
+        else:
+            return HTTPAuthCredential(username=username, password=None)
+
+    def get_password(self, name: str, username: str) -> str | None:
```

Comment:
This method is unused or did I miss something?

Reply:
Yes, i removed it